### PR TITLE
fix #195891: score information too big for small screens

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
and some simplifications and minor corrections to the dialog.

Meant as a possible alternative to #3260, but making the dialog smaller only when running on a too small screen

### CODE REVIEW

**Comment by MarcSabatella** [comment]
General comment when we do these sorts of tweaks: make sure what we do isn't optimized for one specific monitor DPI only.  I mean, 1000 pixels is one size on a 100 DPI monitor but an entirely different size on a 300 DPI monitor or a 72 DPI monitor.  I have no idea what that might mean with respect to this specific PR, just saying, in general, we need to be DPI agnostic and not rely on absolute pixel counts in places where that is not a good measure.

**Comment by Jojo-Schmitz** [comment]
The problematic sizes at hand here are not likely found on high DPI screens or are they?

**Comment by MarcSabatella** [comment]
Probably not.  I'm just saying, adding more code that relies on specific pixel counts should be done only if there is no other choice.  We don't want to get in the habit of thinking 1000 pixels is a meaningful measure across different systems.  We're focusing here on screens of different pixel dimensions but not screens of different pixel densities.  Ideally we'd have an overall strategy and use it consistently, so we'd basically always be doing the right thing.

But for example, if we were so naive as to try to set a dialog size to exactly 600 pixels high because that happens to be big enough to fit everything in on a 100 DPI monitor, people with 300 DPI monitors will be complaining the dialog is too small.  Again, not saying that is happening here, just a reminder that we need to keep this in mind at all times when dealing with these sorts of code changes.

**Comment by Jojo-Schmitz** [comment]
Taking guiScaling into account then should make it save?

**Comment by Jojo-Schmitz** [comment]
Another issue with that dialog: it has a checkbox `Sign out on exit`, but this is set invisble, so not used at all currently.
When should it be used, on pressing OK, or on Cancel too or on any close of that dialog? Or only on upload success? Or on exiting MuseScore?

**Comment by MarcSabatella** [comment]
Yes, in general, taking scaling into account is one way to make a design more responsive to different resolutions.  But as far as I know, this shouldn't be necessary with respect to dialog sizes.  What *should* happen is the dialog sizes itself automatically based on the contents; there shouldn't normally need to be any place where we set absolute dimensions.  Unfortunately that is also what leads to problems such as what this PR is trying to fix - a dialog simply has too much stuff in it to fit on small / low resolution screens.  I'm not saying I know an easy answer.  I think in general setting a "minimimum* size should be safe for high DPI screen - the dialog will automaticaly be bigger as needed.

File: mscore/uploadscoredialog.cpp [hunk 1]
```diff
@@ -233,6 +233,10 @@ void UploadScoreDialog::onGetScoreSuccess(const QString &t, const QString &desc,
       // file with score info
       title->setText(t);
       description->setPlainText(desc);
+      QScreen* screen      = QGuiApplication::primaryScreen();
```
Comment (lasconic) [inline mscore/uploadscoredialog.cpp]: As explained by Marc, this is too much.

Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: What exactly is too much?

Comment (lasconic) [inline mscore/uploadscoredialog.cpp]: Getting the screen, compute values depending on screen size etc...

Comment (Jojo-Schmitz) [inline mscore/uploadscoredialog.cpp]: How else to shrink the Dialog dependant on screen height?

Comment (MarcSabatella) [inline mscore/uploadscoredialog.cpp]: Unfortunately I don't know what the strategy *should* be, I just know some things that *don't* tend to work well.

**Pull request closed without merging**